fix(parser): correct the method's `FormalParameterKind` to `UniqueFormalParameters`

Dunqing opened the pull request:
https://tc39.es/ecma262/#sec-method-definitions-static-semantics-early-errors

15.4.1 Static Semantics: Early Errors
[MethodDefinition](https://tc39.es/ecma262/#prod-MethodDefinition) : [ClassElementName](https://tc39.es/ecma262/#prod-ClassElementName) ( [UniqueFormalParameters](https://tc39.es/ecma262/#prod-UniqueFormalParameters) ) { [FunctionBody](https://tc39.es/ecma262/#prod-FunctionBody) }
It is a Syntax Error if [FunctionBodyContainsUseStrict](https://tc39.es/ecma262/#sec-static-semantics-functionbodycontainsusestrict) of [FunctionBody](https://tc39.es/ecma262/#prod-FunctionBody) is true and [IsSimpleParameterList](https://tc39.es/ecma262/#sec-static-semantics-issimpleparameterlist) of [UniqueFormalParameters](https://tc39.es/ecma262/#prod-UniqueFormalParameters) is false.
It is a Syntax Error if any element of the [BoundNames](https://tc39.es/ecma262/#sec-static-semantics-boundnames) of [UniqueFormalParameters](https://tc39.es/ecma262/#prod-UniqueFormalParameters) also occurs in the [LexicallyDeclaredNames](https://tc39.es/ecma262/#sec-static-semantics-lexicallydeclarednames) of [FunctionBody](https://tc39.es/ecma262/#prod-FunctionBody).

As you can see, the parameters of the method definition should be `UniqueFormalParameters`.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10162?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10162** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10162?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10162?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10161** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10161?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#10160** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10160?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10162).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Apr 1, 9:05 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(parser): correct the method's `FormalParameterKind` to `UniqueFormalParameters` (#10162)

https://tc39.es/ecma262/#sec-method-definitions-static-semantics-early-errors

15.4.1 Static Semantics: Early Errors
[MethodDefinition](https://tc39.es/ecma262/#prod-MethodDefinition) : [ClassElementName](https://tc39.es/ecma262/#prod-ClassElementName) ( [UniqueFormalParameters](https://tc39.es/ecma262/#prod-UniqueFormalParameters) ) { [FunctionBody](https://tc39.es/ecma262/#prod-FunctionBody) }
It is a Syntax Error if [FunctionBodyContainsUseStrict](https://tc39.es/ecma262/#sec-static-semantics-functionbodycontainsusestrict) of [FunctionBody](https://tc39.es/ecma262/#prod-FunctionBody) is true and [IsSimpleParameterList](https://tc39.es/ecma262/#sec-static-semantics-issimpleparameterlist) of [UniqueFormalParameters](https://tc39.es/ecma262/#prod-UniqueFormalParameters) is false.
It is a Syntax Error if any element of the [BoundNames](https://tc39.es/ecma262/#sec-static-semantics-boundnames) of [UniqueFormalParameters](https://tc39.es/ecma262/#prod-UniqueFormalParameters) also occurs in the [LexicallyDeclaredNames](https://tc39.es/ecma262/#sec-static-semantics-lexicallydeclarednames) of [FunctionBody](https://tc39.es/ecma262/#prod-FunctionBody).

As you can see, the parameters of the method definition should be `UniqueFormalParameters`.
